### PR TITLE
Add methods connectioninterface

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | ❌

I am adding the methods that PSALM indicates to me in the implementations, so we do not add unused methods.

### CODE REVIEW

File: src/Connection/ConnectionInterface.php [hunk 1]
```diff
@@ -24,10 +25,78 @@ interface ConnectionInterface
...
+     * `;charset=UTF-8` to the DSN string.
+     *
+     * The same applies for if you're using GBK or BIG5 charset with MySQL, then it's highly recommended to specify
+     * charset via {@see dsn} like `'mysql:dbname=mydatabase;host=127.0.0.1;charset=GBK;'`.
+     *
+     * @param string|null $value
+     *
+     * @return void
+     */
+    public function setCharset(?string $value): void;
```
Comment (samdark) [inline src/Connection/ConnectionInterface.php:91]: Do we need to set it in runtime?

File: src/Connection/ConnectionInterface.php [hunk 2]
```diff
@@ -24,10 +25,78 @@ interface ConnectionInterface
...
+
+    /**
+     * Whether to enable read/write splitting by using {@see setSlaves()} to read data. Note that if {@see setSlaves()}
+     * is empty, read/write splitting will NOT be enabled no matter what value this property takes.
+     *
+     * @param bool $value
+     *
+     * @return void
+     */
+    public function setEnableSlaves(bool $value): void;
```
Comment (samdark) [inline src/Connection/ConnectionInterface.php:101]: Do we need to set it in runtime?

Comment (terabytesoftw) [reply src/Connection/ConnectionInterface.php:101]: Error in `yiisoft/yii-db-migration`.

```php
ERROR: UndefinedInterfaceMethod - src\Migration.php:59:20 - Method Yiisoft\Db\Co nnection\ConnectionInterface::setEnableSlaves does not exist (see https://psalm. dev/181)                                                                                 $this->db->setEnableSlaves(false);
```

File: src/Connection/ConnectionInterface.php (hunk 1 above)
Comment (terabytesoftw) [reply src/Connection/ConnectionInterface.php:91]: Error in `yiisoft/yii-db-migration`

```php

ERROR: UndefinedInterfaceMethod - src\Service\Database\ListTablesService.php:45: 63 - Method Yiisoft\Db\Connection\ConnectionInterface::getDSN does not exist (se e https://psalm.dev/181)                                                                 $dbname = $this->getDsnAttribute('dbname', $this->db->getDSN());
```

which refers to charset.

Comment (terabytesoftw) [reply src/Connection/ConnectionInterface.php]: done remove no affect static analisys.

File: src/Connection/ConnectionInterface.php (hunk 2 above)
Comment (samdark) [reply src/Connection/ConnectionInterface.php:84]: Yes, I know. I'm asking if we need this method in both interface and Connection class.

Comment (terabytesoftw) [reply src/Connection/ConnectionInterface.php:84]: Only in the interface, in the connection methods of the implementation I think that we must have the methods that have change by the driver is my opinion, the rest in the abstract class.

**Pull request merged**